2005-05-17  Neal H. Walfield  <neal@gnu.org>

	[BZ #1350]
	* sysdeps/posix/getaddrinfo.c (gaih_local): Check [_HAVE_SA_LEN]
	instead of [SALEN].
	(gaih_inet): Likewise.
2005-02-16  Roland McGrath  <roland@redhat.com>

	* sysdeps/posix/getaddrinfo.c (gaih_inet): Fix type of ADDR local.

===================================================================
RCS file: /cvs/glibc/libc/sysdeps/posix/getaddrinfo.c,v
retrieving revision 1.85
retrieving revision 1.85.2.1
diff -u -r1.85 -r1.85.2.1
--- libc/sysdeps/posix/getaddrinfo.c	2004/11/22 18:41:37	1.85
+++ libc/sysdeps/posix/getaddrinfo.c	2005/10/16 10:15:30	1.85.2.1
@@ -196,10 +196,10 @@
   (*pai)->ai_addrlen = sizeof (struct sockaddr_un);
   (*pai)->ai_addr = (void *) (*pai) + sizeof (struct addrinfo);
 
-#if SALEN
+#ifdef _HAVE_SA_LEN
   ((struct sockaddr_un *) (*pai)->ai_addr)->sun_len =
     sizeof (struct sockaddr_un);
-#endif /* SALEN */
+#endif /* _HAVE_SA_LEN */
 
   ((struct sockaddr_un *)(*pai)->ai_addr)->sun_family = AF_LOCAL;
   memset(((struct sockaddr_un *)(*pai)->ai_addr)->sun_path, 0, UNIX_PATH_MAX);
@@ -680,7 +680,7 @@
 				    h->h_length);
 			  else
 			    {
-			      int32_t *addr = (uint32_t *) (*pat)->addr;
+			      uint32_t *addr = (uint32_t *) (*pat)->addr;
 			      addr[3] = *(uint32_t *) h->h_addr_list[i];
 			      addr[2] = htonl (0xffff);
 			      addr[1] = 0;
@@ -1076,9 +1076,9 @@
 	    ai->ai_canonname = (char *) canon;
 	    canon = NULL;
 
-#if SALEN
+#ifdef _HAVE_SA_LEN
 	    ai->ai_addr->sa_len = socklen;
-#endif /* SALEN */
+#endif /* _HAVE_SA_LEN */
 	    ai->ai_addr->sa_family = family;
 
 	    if (family == AF_INET6)
